File: ALGORITHMS/Python/Searching/linear_search.py
# Linear search in python

def linear_search (arr , target) :
  
  # iterate loop till you get target.
  
  for i in range(len(arr)):
    
    # check if we found target or not
    
    if(arr[i] == target):    
      
      # return index of currunt element
      
      return i
    
arr = [5,4,2,8,1,8]
target = 4
print( linear_search(arr,target))

File: ALGORITHMS/Cpp/Searching/linearSearch.cpp
// Linear search in cpp

#include<iostream>

using namespace std;

// Linear search function

int linearSearch(int *arr, int length_arr, int target)
{
    // Iterate loop length_arr times
    
    for(int i = 0 ; i < length_arr ; i++)
    {
        // Check wheather target is present or not
      
        if(arr[i] == target)
        {   
            //return index of traget
          
            return i;
        }
    }
  return -1;
}

int main(){

    int arr[10] = {2, 4, 1, 8, 79, 45, 55, 12, 3, 7};
  
    int length_arr = 10;
  
    int target = 7;
  
    // Call linearSearch function
  
    int ans = linearSearch(arr, length_arr, target);
  
    if(ans == -1)
    {
        cout<<"target is not prensent in an Array ";
    }
    else
    {
        cout<<endl<<"target is  prensent in an Array at index : "<<ans;   
    }
    return 0;
}
